lib/beaker-hostgenerator/parser.rb

```diff
@@ -14,7 +14,28 @@ module BeakerHostGenerator
   # each token you would call `is_ostype_token?` and/or `parse_node_info_token`.
   module Parser
 
-    # Capture role and bit width information about the node.
+    # Parses a single node definition into the following components:
+    #
+    #   * bits             Uppercase-only alphanumeric
+    #                      Examples: 32, 64, POWER7 (note 7POWER is invalid)
```

Comment:
Ah, `S390X` wouldn't work with this, you're right.  I don't think there's any reason we can't improve this as you suggest, I'll make that happen.

Glad you caught this.